DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -265,10 +246,19 @@ public void PromoteDelegateToAdmin(AdminRoles adminRoles, int? categoryId, int d
                 adminRoles.IsContentCreator,
                 adminRoles.IsCmsAdministrator,
                 adminRoles.IsCmsManager,
-                delegateUser.ProfileImage
+                delegateUser.JobGroupId
             );
 
-            var userId = userDataService.GetUserIdFromDelegateId(delegateId);
+            if (admin != null)
+            {
+                if (admin.Active)
+                {
+                    throw new AdminCreationFailedException(AdminCreationError.ActiveAdminAlreadyExists);
+                }
+
+                userDataService.UpdateAdminAccount(adminRegistrationModel, admin.Id);
+            }
+
             registrationDataService.RegisterAdmin(adminRegistrationModel, userId);

```

Comment:
This is getting called all the time, regardless as to if an account already exists. This needs to be in an `else` statement. I would assume that this would fail if this was manually tested? If not, we probably need to chuck some extra handling in this method.